libbeat/otelbeat/converters/beatconverter.go

```diff
@@ -0,0 +1,106 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package converters
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/confmap"
+	"go.uber.org/zap"
+
+	"github.com/elastic/beats/v7/libbeat/outputs/elasticsearch"
+	"github.com/elastic/elastic-agent-libs/config"
+)
+
+// list of supported beatreceivers
+var supportedReceivers = []string{"filebeatreceiver", "metricbeatreceiver"} // Add more beat receivers to this list when we add support
+
+type converter struct {
+	logger *zap.Logger
+}
+
+// NewFactory returns a factory for a  confmap.Converter,
+func NewFactory() confmap.ConverterFactory {
+	return confmap.NewConverterFactory(newConverter)
+}
+
+func newConverter(set confmap.ConverterSettings) confmap.Converter {
+	return converter{
+		logger: set.Logger,
+	}
+}
+
+// [beatreceiver].output is unpacked to OTel config here
+func (c converter) Convert(_ context.Context, conf *confmap.Conf) error {
+
+	for _, receiverbeat := range supportedReceivers {
+		var out map[string]any
+
+		// check if supported beat receiver is configured. Skip translation logic if not
+		if v := conf.Get("receivers::" + receiverbeat); v == nil {
+			continue
+		}
+
+		receiverCfg, _ := conf.Sub("receivers::" + receiverbeat)
+		outputs, _ := receiverCfg.Sub("output")
+
+		for key := range outputs.ToStringMap() {
```

Comment:
I was thinking we could rely on that kind of error handling on filebeatreceiver to keep this logic here straightforward.

We just want to make sure that given input (even if incorrect is properly translated). wdyt

Reply:
> Did you mean outputs?

yeah, my bad, I meant outputs.

> Also if fbreceiver does support multiple outputs can we allow beats users to configure multiple outputs in the "otel" mode? This will only be a feature addition. Any thoughts?

The receiver itself (internally) does not support multiple outputs, it still works with the single output – otelconsumer. If we want to translate the current Beats configuration to OTel we definitely don't need support for multiple outputs.

I don't have a strong opinion about removing multiple output support from this PR but it's out of the scope of this task and extends current functionality of Beats which will require some additional documentation work. Not sure we want to do it now.